crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -337,24 +346,56 @@ impl SyncTimelineEvent {
         event: Raw<AnySyncTimelineEvent>,
         push_actions: Vec<Action>,
     ) -> Self {
-        Self { event, encryption_info: None, push_actions, unsigned_encryption_info: None }
+        Self {
+            inner_event: event,
+            inner_encryption_info: None,
+            push_actions,
+            unsigned_encryption_info: None,
+        }
     }
 
     /// Get the event id of this `SyncTimelineEvent` if the event has any valid
     /// id.
     pub fn event_id(&self) -> Option<OwnedEventId> {
-        self.event.get_field::<OwnedEventId>("event_id").ok().flatten()
+        self.inner_event.get_field::<OwnedEventId>("event_id").ok().flatten()
+    }
+
+    /// Returns a reference to the (potentially decrypted) Matrix event inside
+    /// this `TimelineEvent`.
+    pub fn event(&self) -> &Raw<AnySyncTimelineEvent> {
+        &self.inner_event
+    }
+
+    /// If the event was a decrypted event that was successfully decrypted, get
+    /// its encryption info. Otherwise, `None`.
+    pub fn encryption_info(&self) -> Option<&EncryptionInfo> {
+        self.inner_encryption_info.as_ref()
+    }
+
+    /// Takes ownership of this `TimelineEvent`, returning the (potentially
+    /// decrypted) Matrix event within.
+    pub fn into_event(self) -> Raw<AnySyncTimelineEvent> {
+        self.inner_event
+    }
```

Comment:
This should be renamed `into_inner` for consistency with the other method below, or `into_raw` if you follow my above suggestion.

Reply:
3a0673f69 renames to `into_raw`